pkg/repository/mirror.go

```diff
@@ -205,11 +213,24 @@ func (l *httpMirror) Download(resource, targetDir string) error {
 
 // Fetch implements the Mirror interface
 func (l *httpMirror) Fetch(resource string, maxSize int64) (io.ReadCloser, error) {
-	return l.download(l.prepareURL(resource), "", maxSize)
+	r, err := l.download(l.prepareURL(resource), filepath.Join(l.tmpDir, resource), maxSize)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer r.Close()
+	path := filepath.Join(l.tmpDir, resource)
+	file, err := os.OpenFile(path, os.O_RDONLY, os.ModePerm)
```

Comment:
It seems the `file` references to the same under-level file `r` returned from `l.download`.